modules/auxiliary/scanner/nfs/nfsmount.rb

```diff
@@ -10,74 +10,101 @@ class MetasploitModule < Msf::Auxiliary
+    )
+  end
+
+  def can_mount?(locations)
+    # attempts to validate if we'll be able to open it or not based on:
+    # 1. its a wildcard, thus we can open it
+    # 2. hostname isn't blank and its in the list
+    # 3. our IP is explicitly listed
+    # 4. theres a CIDR notation that we're included in.
+    return true unless datastore['Mountable']
+    return true if locations.include? '*' ||
+                                      (locations.include?(datastore['HOSTNAME']) && !datastore['HOSTNAME'].blank?) ||
+                                      locations.include?(datastore['LHOST'])
+
+    locations.each do |location|
+      # if it has a subnet mask, convert it to cidr
+      if %r{(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/)(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})} =~ location
+        location = "#{Regexp.last_match(1)}#{Rex::Socket.addr_atoc(Regexp.last_match(2))}"
+      end
+      return true if Rex::Socket::RangeWalker.new(location).include?(datastore['LHOST'])
+    end
+    false
   end
 
   def run_host(ip)
+    program	= 100005
+    progver	= 1
+    procedure	= 5
 
-    begin
-      program		= 100005
-      progver		= 1
-      procedure	= 5
+    sunrpc_create(datastore['PROTOCOL'], program, progver)
+    sunrpc_authnull
+    resp = sunrpc_call(procedure, '')
 
-      sunrpc_create(datastore['PROTOCOL'], program, progver)
-      sunrpc_authnull()
-      resp = sunrpc_call(procedure, "")
+    # XXX: Assume that transport is udp and port is 2049
+    # Technically we are talking to mountd not nfsd
 
-      # XXX: Assume that transport is udp and port is 2049
-      # Technically we are talking to mountd not nfsd
+    report_service(
+      host: ip,
+      proto: datastore['PROTOCOL'],
+      port: 2049,
+      name: 'nfsd',
+      info: "NFS Daemon #{program} v#{progver}"
+    )
 
-      report_service(
-        :host  => ip,
-        :proto => datastore['PROTOCOL'],
-        :port  => 2049,
-        :name  => 'nfsd',
-        :info  => "NFS Daemon #{program} v#{progver}"
-      )
+    exports = resp[3, 1].unpack('C')[0]
+    if (exports == 0x01)
+      shares = []
+      while Rex::Encoder::XDR.decode_int!(resp) == 1
+        dir = Rex::Encoder::XDR.decode_string!(resp)
+        grp = []
+        grp << Rex::Encoder::XDR.decode_string!(resp) while Rex::Encoder::XDR.decode_int!(resp) == 1
 
-      exports = resp[3,1].unpack('C')[0]
-      if (exports == 0x01)
-        shares = []
-        while Rex::Encoder::XDR.decode_int!(resp) == 1 do
-          dir = Rex::Encoder::XDR.decode_string!(resp)
-          grp = []
-          while Rex::Encoder::XDR.decode_int!(resp) == 1 do
-            grp << Rex::Encoder::XDR.decode_string!(resp)
-          end
-          print_good("#{ip} NFS Export: #{dir} [#{grp.join(", ")}]")
-          shares << [dir, grp]
+        if can_mount? grp
+          print_good("#{ip} NFS Export: #{dir} [#{grp.join(', ')}]")
+        else
+          print_status("#{ip} NFS Export: #{dir} [#{grp.join(', ')}]")
         end
```

Comment:
I like that not-mountable NFS shares are still printed. Also printing shares as they're discovered is useful.

There's no differentiation in output between a mountable and not-mountable share, beyond the good/status prefix. That's kind of ambiguous. The data stored with `report_note` also does not appear to differentiate the two.

Perhaps printing a table once complete would be useful? The output could include a column to show whether each entry is mountable and the table could be stored as CSV with `store_loot` (for easy parsing and to support msf users who don't use a database). A table could be easily extended. For example, the module could also optionally store whether an export is writable.